code/datums/status_effects/debuffs/slimed.dm

```diff
@@ -0,0 +1,69 @@
+/// The minimum amount of water stacks needed to start washing off the slime.
+#define MIN_WATER_STACKS 5
+/// The minimum amount of health a mob has to have before the status effect is removed.
+#define MIN_HEALTH 10
+
+/atom/movable/screen/alert/status_effect/slimed
+	name = "Covered in Slime"
+	desc = "You are covered in slime and it's eating away at you! Find a way to wash it off!"
+	icon_state = "slimed"
+
+/datum/status_effect/slimed
+	id = "slimed"
+	tick_interval = 3 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/slimed
+	remove_on_fullheal = TRUE
+
+	/// The amount of slime stacks that were applied, reduced by showering yourself in water.
+	var/slime_stacks = 10 // ~10 seconds of standing under a shower
+
+/datum/status_effect/slimed/on_apply()
+	if(owner.get_organic_health() <= MIN_HEALTH)
+		return FALSE
+	owner.visible_message(
+		span_warning("[owner] has been covered in a thick layer of slime!"),
+		span_userdanger("You have been covered in a thick layer of slime! Find a way to wash it off!"))
+	return ..()
+
+/datum/status_effect/slimed/tick(seconds_per_tick)
+	// remove from the mob once we have dealt enough damage
+	if(owner.get_organic_health() <= MIN_HEALTH)
+		owner.visible_message(
+			span_warning("The layer of slime slowly drips off [owner] as it leaves [owner.p_their()] weakened body behind."),
+			span_warning("You feel the layer of slime crawling off of your weakened body."))
+		qdel(src)
+		return
+
+	// handle washing slime off
+	var/datum/status_effect/fire_handler/wet_stacks/wetness = locate() in owner.status_effects
+	if(istype(wetness) && wetness.stacks > (MIN_WATER_STACKS * seconds_per_tick))
```

Comment:
I wonder if this effect as a whole should be a subtype of `fire_handler`, it's my understanding that you can set a `enemy_type` which handles much of this behavior for you

Reply:
`fire_handler` seems to be a bit too hardcoded to work with fire for me to be able to work with its implementation, sadly

Reply:
Not throwing shade but how does wet stacks work if fire handlers are too hardcoded to work with fire

Reply:
you make a good point

Reply:
Could just set the stack modifier to 0

Reply:
that feels like a big hack, id be more comfortable rewriting this into a more general superclass tbh
let me know if this is blocking enough to do it in this PR

Reply:
Also, then I would lose the ability to remove water stacks + divide by zero

https://github.com/distributivgesetz/tgstation/blob/f872ab8363737447c51cfbacf47e4461f6e30733/code/datums/status_effects/debuffs/fire_stacks.dm#L41-L56